### PR TITLE
fix: add protected-file exclusion to issue-creating agents (#927)

### LINKED ISSUE #927: bug: code-health agent creates issues requiring protected file changes, causing implementer to loop
## Problem

The code-health agent has no awareness of protected files. It analyzes the entire codebase including `.github/` and flags cleanup opportunities there (e.g., empty `PRODUCT_VISION.md`).

When the implementer picks up these issues, it creates a branch, pushes code, but the `safe_outputs` system blocks the PR because the changes touch protected paths. It falls back to creating a "review issue" instead of a PR. The orchestrator then re-dispatches the implementer, which fails the same way — 3 times for issue #671.

### Evidence

Issue #671 asked the agent to populate `.github/PRODUCT_VISION.md`. The implementer attempted it 3 times:
- Run 23923589675 → review issue #675 (protected-file-protection fallback)
- Run 23945169580 → review issue #692 (same)
- Run 23989981998 → review issue #741 (same)

Log line from safe_outputs:
```
Created protected-file-protection review issue #675
Code push `create_pull_request` fell back to review issue #675
```

### Root Cause

`code-health.md` instructions say "Read all files in the repository" and "open issues for anything worth fixing" with no exclusion for protected paths. The agent does not know which files the implementer will be blocked from modifying.

## Fix

Add an instruction to `code-health.md` telling it not to open issues that require modifying files in protected paths (`.github/workflows/`, `.github/copilot-instructions.md`, `.github/CODING_GUIDELINES.md`, etc.). These changes require human intervention and waste pipeline resources when dispatched.

## Related

- #671 — the issue that triggered 3 failed attempts
- #675, #692, #741 — the 3 review issues created by protected-file fallback

### PR DESCRIPTION
**Pull request by microsasa**
## Problem

The code-health, perf-analysis, test-analysis, and feature-planner agents had no awareness of protected files. They could file issues requiring changes to paths the implementer cannot touch (`.github/`, `pyproject.toml`, etc.), causing the implementer to loop — pushing code, hitting safe-outputs, falling back to a review issue, repeat. Issue #671 triggered 3 failed attempts this way.

## Changes

All 4 issue-creating agents now include protected-file exclusion instructions:

- **code-health.md** — do not open issues requiring changes to `.github/`, `.agents/`, `pyproject.toml`, or `uv.lock`
- **perf-analysis.md** — same
- **test-analysis.md** — same
- **feature-planner.md** — same, plus a **dependency-aware** flow: if a feature needs new deps, file only a `[aw][deps]` issue (labeled `auto-feature` only, not `aw`) so the orchestrator won't dispatch it. A maintainer installs the dep manually, closes the issue, and the next planner run files the actual feature.

Wording is normalized across all 4 files. Lock files recompiled.

## Review feedback incorporated

Three-model review (Opus 4.6, Codex, Sonnet 4.6) surfaced and we addressed:
- Contradiction between "no protected files" and "dependency-aware" bullets → added carve-out
- "merged" wording → changed to "after a maintainer resolves and closes"
- Deps issue labeled `aw` would get dispatched to implementer → dropped `aw`, kept `auto-feature` only
- `*.lock.yml` redundant with `.github/` → removed

Closes #927

### CODE REVIEW

File: .github/workflows/feature-planner.md [hunk 1]
```diff
@@ -46,6 +46,8 @@ Compare the current codebase against the vision. Identify one small, concrete st
 - **Specific enough to code** — include the files to create or modify, the behavior to add, and how to verify it works.
 - **Non-conflicting** — do not propose changes that would contradict or duplicate any open issue.
 - **Incremental** — prefer the smallest useful step over an ambitious leap. Each step should leave the codebase in a working, releasable state.
+- **No protected files** — do not propose features that require changes to `.github/`, `.agents/`, `pyproject.toml`, `uv.lock`, or other dependency/build config files — except as described in the Dependency-aware rule below. The implementer agent cannot create PRs that touch these paths.
```
[the remark this answers is not in the document]
Comment (microsasa) [reply .github/workflows/feature-planner.md]: Good catch — `.agents/` doesn't exist at the repo root; it's `.github/agents/`, already covered by the `.github/` prefix. Removed `.agents/` from all 4 files in 425f9f8.

File: .github/workflows/code-health.md [hunk 2]
```diff
@@ -38,4 +38,4 @@ Read all files in the repository. Read all open issues in the repository. Identi
...
+Do not open issues for things already caught by CI (ruff, pyright, bandit). Do not open issues for things that already have an open issue. Do not open an issue that is just a nit — if there are many small nits that together form a meaningful cleanup, bundle them into one issue. Do not open issues for performance problems (those belong to perf-analysis). Do not open issues that require modifying protected files — the implementer agent cannot create PRs that touch these paths. Protected paths include: `.github/`, `.agents/`, `pyproject.toml`, and `uv.lock`. If you find more issues than you can open, prioritize by severity: prefer bugs and correctness issues over cosmetic cleanup. If nothing worth fixing is found, do not create any issues.
```
[the remark this answers is not in the document]
Comment (microsasa) [reply .github/workflows/code-health.md]: Good catch — `.agents/` doesn't exist at the repo root; it's `.github/agents/`, already covered by the `.github/` prefix. Removed `.agents/` from all 4 files in 425f9f8.

File: .github/workflows/perf-analysis.md [hunk 3]
```diff
@@ -40,4 +40,4 @@ Look for performance problems such as algorithmic inefficiency (O(n²) loops, re
...
 
-Do not open issues for micro-optimizations that save nanoseconds. Do not open issues for things already caught by CI (ruff PERF rules). Do not open issues for things that already have an open issue. If you find more issues than you can open, prioritize by impact: prefer findings in hot paths, high-frequency call sites, or code that scales with input size. If nothing worth optimizing is found, do not create any issues.
+Do not open issues for micro-optimizations that save nanoseconds. Do not open issues for things already caught by CI (ruff PERF rules). Do not open issues for things that already have an open issue. Do not open issues that require modifying protected files — the implementer agent cannot create PRs that touch these paths. Protected paths include: `.github/`, `.agents/`, `pyproject.toml`, and `uv.lock`. If you find more issues than you can open, prioritize by impact: prefer findings in hot paths, high-frequency call sites, or code that scales with input size. If nothing worth optimizing is found, do not create any issues.
```
[the remark this answers is not in the document]
Comment (microsasa) [reply .github/workflows/perf-analysis.md]: Good catch — `.agents/` doesn't exist at the repo root; it's `.github/agents/`, already covered by the `.github/` prefix. Removed `.agents/` from all 4 files in 425f9f8.

File: .github/workflows/test-analysis.md [hunk 4]
```diff
@@ -38,4 +38,4 @@ Read all files in the repository. Read all open issues in the repository. Identi
...
 
-Do not suggest trivial tests or tests that duplicate existing coverage. Do not open issues for things that already have an open issue. Do not open an issue that is just a nit — if there are many small gaps that together form a meaningful improvement, bundle them into one issue. If you find more issues than you can open, prioritize by risk: prefer coverage gaps in core business logic over edge-case branches. If the test suite is already comprehensive, do not create any issues.
+Do not suggest trivial tests or tests that duplicate existing coverage. Do not open issues for things that already have an open issue. Do not open an issue that is just a nit — if there are many small gaps that together form a meaningful improvement, bundle them into one issue. Do not open issues that require modifying protected files — the implementer agent cannot create PRs that touch these paths. Protected paths include: `.github/`, `.agents/`, `pyproject.toml`, and `uv.lock`. If you find more issues than you can open, prioritize by risk: prefer coverage gaps in core business logic over edge-case branches. If the test suite is already comprehensive, do not create any issues.
```
[the remark this answers is not in the document]
Comment (microsasa) [reply .github/workflows/test-analysis.md]: Good catch — `.agents/` doesn't exist at the repo root; it's `.github/agents/`, already covered by the `.github/` prefix. Removed `.agents/` from all 4 files in 425f9f8.
